### PR TITLE
Sandbox CLI

### PR DESCRIPTION
**Pull request by feakuru**

### CODE REVIEW

**Comment by feakuru** [comment]
[Sandbox CLI](https://trello.com/c/P11kt3I9/13-sandbox-cli)

File: src/pytezos/cli/cli.py [hunk 1]
```diff
@@ -22,6 +28,8 @@
     "codemirror_mode": "michelson",
 }
 
+SANDBOX_IMAGE = 'bakingbad/sandboxed-node:v9.0-rc1-1'
```
Comment (droserasprout) [inline src/pytezos/cli/cli.py:31]: User may want to use it's own image, it's better to make this a CLI option

File: src/pytezos/cli/cli.py [hunk 2]
```diff
@@ -173,5 +185,76 @@ def deploy(
             logger.info(status)
 
 
+@cli.command(help='Run containerized sandbox')
+@click.option('--protocol', type=click.Choice(['Florence', 'Edo']), help='Protocol to use', default='Florence')
```
Comment (droserasprout) [inline src/pytezos/cli/cli.py:189]: I suggest to use multiple aliases here and pass value as is if none matched. @m-kus, could we call `edo2` protocol just `edo`?

**Review by droserasprout — CHANGES_REQUESTED** [review]

File: src/pytezos/cli/cli.py (hunk 1 above)
Comment (feakuru) [reply src/pytezos/cli/cli.py:31]: You mean just the tag or the image name too?

Comment (droserasprout) [reply src/pytezos/cli/cli.py:31]: Both image and tag.

Comment (feakuru) [reply src/pytezos/cli/cli.py:31]: Done

File: src/pytezos/cli/cli.py (hunk 2 above)
Comment (m-kus) [reply src/pytezos/cli/cli.py:190]: Yes, we can

**Review by droserasprout — APPROVED** [review]

File: src/pytezos/cli/cli.py [hunk 3]
```diff
@@ -173,5 +185,78 @@ def deploy(
...
+    )
+
+    atexit.register(node.stop)
+
+    logger.info('Giving node 5 seconds to start.')
+
+    def end_waiting():
+        logger.info('Waiting over.')
+
+    waiting_for_node_start_thread = threading.Timer(5, end_waiting)
```
Comment (m-kus) [inline src/pytezos/cli/cli.py:229]: 5 seconds is too much, better poll /version endpoint as it's done in sandbox testcase

**Comment by m-kus** [comment]
[Sandbox CLI: start as early as possible](https://trello.com/c/RnUBV4oj/57-sandbox-cli-start-as-early-as-possible)

File: src/pytezos/cli/cli.py [hunk 4]
```diff
@@ -173,5 +176,43 @@ def deploy(
...
+        'florence': FLORENCE,
+    }[protocol]
+
+    SandboxedNodeTestCase.PROTOCOL = protocol
+    SandboxedNodeTestCase.IMAGE = image
+    SandboxedNodeTestCase.PORT = port
+    SandboxedNodeTestCase.setUpClass()
+
+    blocks_baked = 0
+    while True:
```
Comment (m-kus) [inline src/pytezos/cli/cli.py:205]: Graceful shutdown by Ctrl+C || SIGTERM

**Pull request merged**